src/main/java/org/broadinstitute/hellbender/tools/sv/SVCallRecord.java

```diff
@@ -0,0 +1,217 @@
+        final int end = variant.getAttributeAsInt(VCFConstants.END_KEY, variant.getStart());
+        final StructuralVariantType type = variant.getStructuralVariantType();
+        final List<String> algorithms = variant.getAttributeAsStringList(GATKSVVCFConstants.ALGORITHMS_ATTRIBUTE, GATKSVVCFConstants.DEPTH_ALGORITHM);
+        final String strands = variant.getAttributeAsString(GATKSVVCFConstants.STRANDS_ATTRIBUTE, "0");
+        if (strands.length() != 2) {
+            throw new IllegalArgumentException("Strands field is not 2 characters long");
+        }
+        final String startStrandChar = strands.substring(0, 1);
+        if (!startStrandChar.equals(SVCallRecordCodec.STRAND_PLUS) && !startStrandChar.equals(SVCallRecordCodec.STRAND_MINUS)) {
+            throw new IllegalArgumentException("Valid start strand not found");
+        }
+        final String endStrandChar = strands.substring(1, 2);
+        if (!endStrandChar.equals(SVCallRecordCodec.STRAND_PLUS) && !endStrandChar.equals(SVCallRecordCodec.STRAND_MINUS)) {
+            throw new IllegalArgumentException("Valid end strand not found");
+        }
+        final boolean startStrand = startStrandChar.equals(SVCallRecordCodec.STRAND_PLUS);
+        final boolean endStrand = endStrandChar.equals(SVCallRecordCodec.STRAND_PLUS);
+        final int length = variant.getAttributeAsInt(GATKSVVCFConstants.SVLEN, 0);
+        return new SVCallRecord(startContig, start, startStrand, endContig, end, endStrand, type, length, algorithms, variant.getGenotypes());
+    }
+
+    public static SVCallRecord createDepthOnlyFromGCNV(final VariantContext variant, final double minQuality) {
+        Utils.nonNull(variant);
+        final List<Genotype> passing = variant.getGenotypes().stream()
+                .filter(Genotype::isCalled)
+                .filter(g -> Integer.valueOf((String)g.getExtendedAttribute(GermlineCNVSegmentVariantComposer.QS)) >= minQuality)
+                .collect(Collectors.toList());
+        if (passing.isEmpty()) return null;
+        final List<String> algorithms = Collections.singletonList(GATKSVVCFConstants.DEPTH_ALGORITHM);
+
+        //TODO : use new vcfs to get actual allele
+        final int copyNumber = Integer.valueOf((String)variant.getGenotypes().get(0).getExtendedAttribute(GermlineCNVSegmentVariantComposer.CN));
+        if (copyNumber == 2) return null;
+        final boolean isDel = copyNumber < 2;
+        final boolean startStrand = isDel ? true : false;
+        final boolean endStrand = isDel ? false : true;
+        final StructuralVariantType type = isDel ? StructuralVariantType.DEL : StructuralVariantType.DUP;
+
+        final String startContig = variant.getContig();
+        final int start = variant.getStart();
+        final int end = variant.getEnd() + 1; // TODO this is a bug with gCNV vcf generation
+        final int length = end - start;
```

Comment:
Is the length inclusive?  That's typically the GATK convention, in which case this needs a +1.  Unless it comes as the above compensatory +1?